neuro_san/session/async_http_service_agent_session.py

```diff
@@ -0,0 +1,107 @@
+
+# Copyright (C) 2023-2025 Cognizant Digital Business, Evolutionary AI.
+# All Rights Reserved.
+# Issued under the Academic Public License.
+#
+# You can be released from the terms, and requirements of the Academic Public
+# License by purchasing a commercial license.
+# Purchase of a commercial license is mandatory for any use of the
+# neuro-san SDK Software in commercial settings.
+#
+# END COPYRIGHT
+
+from typing import Any
+from typing import Dict
+from typing import Generator
+
+import json
+
+from aiohttp import ClientSession
+
+from neuro_san.interfaces.async_agent_session import AsyncAgentSession
+from neuro_san.session.abstract_http_service_agent_session import AbstractHttpServiceAgentSession
+
+
+class AsyncHttpServiceAgentSession(AbstractHttpServiceAgentSession, AsyncAgentSession):
+    """
+    Implementation of AsyncAgentSession that talks to an HTTP service.
+    """
+
+    async def function(self, request_dict: Dict[str, Any]) -> Dict[str, Any]:
+        """
+        :param request_dict: A dictionary version of the FunctionRequest
+                    protobufs structure. Has the following keys:
+                        <None>
+        :return: A dictionary version of the FunctionResponse
+                    protobufs structure. Has the following keys:
+                "function" - the dictionary description of the function
+        """
+        path: str = self.get_request_path("function")
+        result_dict: Dict[str, Any] = None
+        try:
+            async with ClientSession(headers=self.get_headers(),
+                                     conn_timeout=self.timeout_in_seconds,
+                                     read_timeout=self.timeout_in_seconds,
+                                     ) as session:
+                async with session.get(path, json=request_dict) as response:
+                    result_dict = await response.json()
+                    return result_dict
```

Comment:
get() translation from synchronous use of request library is pretty straightforward.